barter/examples/custom_strategy.rs

```diff
@@ -0,0 +1,414 @@
+use barter::{
+    engine::{
+        audit::Audit,
+        clock::{EngineClock, LiveClock},
+        command::Command,
+        run,
+        state::{
+            self,
+            instrument::{
+                filter::InstrumentFilter,
+                market_data::{DefaultMarketData, MarketDataState},
+            },
+            order::manager::OrderManager,
+            trading::TradingState,
+        },
+        Engine, Processor,
+    },
+    execution::builder::ExecutionBuilder,
+    logging::init_logging,
+    risk::{DefaultRiskManager, DefaultRiskManagerState},
+    statistic::time::Daily,
+    strategy::{
+        algo::AlgoStrategy, close_positions::ClosePositionsStrategy,
+        on_disconnect::OnDisconnectStrategy, on_trading_disabled::OnTradingDisabled,
+    },
+    EngineEvent,
+};
+use barter_data::{
+    event::MarketEvent,
+    streams::{
+        builder::dynamic::indexed::init_indexed_multi_exchange_market_stream,
+        reconnect::stream::ReconnectingStream,
+    },
+    subscription::SubKind,
+};
+use barter_execution::{
+    balance::Balance,
+    client::mock::MockExecutionConfig,
+    order::{ClientOrderId, Order, OrderKind, RequestCancel, RequestOpen, StrategyId, TimeInForce},
+    AccountEvent, AccountEventKind,
+};
+use barter_instrument::{
+    asset::AssetIndex,
+    exchange::{ExchangeId, ExchangeIndex},
+    index::IndexedInstruments,
+    instrument::{
+        kind::InstrumentKind,
+        spec::{
+            InstrumentSpec, InstrumentSpecNotional, InstrumentSpecPrice, InstrumentSpecQuantity,
+            OrderQuantityUnits,
+        },
+        Instrument, InstrumentIndex,
+    },
+    Side, Underlying,
+};
+use barter_integration::channel::{mpsc_unbounded, ChannelTxDroppable, Tx};
+use fnv::FnvHashMap;
+use futures::StreamExt;
+use rust_decimal::Decimal;
+use rust_decimal_macros::dec;
+use tracing::{debug, info, warn};
+
+const EXCHANGE: ExchangeId = ExchangeId::BinanceSpot;
+const RISK_FREE_RETURN: Decimal = dec!(0.05);
+const MOCK_EXCHANGE_ROUND_TRIP_LATENCY_MS: u64 = 100;
+const MOCK_EXCHANGE_FEES_PERCENT: Decimal = dec!(0.05);
+const STARTING_BALANCE_USDT: Balance = Balance {
+    total: dec!(10_000.0),
+    free: dec!(10_000.0),
+};
+const STARTING_BALANCE_BTC: Balance = Balance {
+    total: dec!(0),
+    free: dec!(0),
+};
+const STARTING_BALANCE_ETH: Balance = Balance {
+    total: dec!(1.0),
+    free: dec!(1.0),
+};
+const STARTING_BALANCE_SOL: Balance = Balance {
+    total: dec!(10.0),
+    free: dec!(10.0),
+};
+
+/// Engine state used for running the custom strategy
+type EngineState =
+    state::EngineState<DefaultMarketData, BuyAndHoldStrategyState, DefaultRiskManagerState>;
+
+struct BuyAndHoldStrategy {
+    /// Strategy Id
+    id: StrategyId,
+    /// Instrument traded by this strategy
+    instrument: InstrumentIndex,
```

Comment:
I know this is an example, but if this is one of the only "strategy examples" i'd prefer that it trades on multiple instruments. 

Someone who's looking for the first time might think that a strategy only works for a single instrument